### ISSUE TITLE
Budget & Push Notification

### ISSUE THREAD

**Issue by mrahmadt** [cioraneanu/firefly-pico#13]
Hello

I'm working on the "Wife Approval Certification" here :)

it's actually Wife & Kids Approval Certification

How easy to add the following features:

- Budget View only: Family will only see assigned budget to them? how much remaining in this month budget and list of transaction associated to this assigned budget, no add transaction or anything
All transactions are automated for me (SMS 2 Transaction) so I do not want my wife and kids to add transaction manually

- Web Push notifications
I'm running different scripts in the backend for example to detect abnormal transactions, sms to transactions, subscription detection, expense analysis, Currently these scripts runs and send details to telegram
It would be an amazing to actually just send the notification and open firefly-pico

- Sub Login
Basically user name and password (maybe even hard coded in the database)  for firefly-pico that will connect to the same account in FireFly

**Comment by cioraneanu** [comment]
Glad I'm not the only one dealing with that certification 😆 

- I can see adding budgets down the road in Pico, but they are not currently a priority since I don't use them. The idea to configure what budgets you see sounds great.

- Push notifications could be implemented but we need some clearly defined use cases. Clicking each notification should take you to an app page inside Pico that has relevant data / actions to take. Notifications like "Hey, your daily bank import is done" isn't very helpful and can be performed in any other notification app. 

Another thing that might be worth thinking is creating and documenting dedicated Pico URLs in order highlight certain resources.
Ex. `https://pico.example.com/transactions/list?date_start=2024-05-02&date_end=2024-05-02`. You can build these URLs and send to in a notification via Telegram, Discord, Ntfy etc.

- Implementing users for Pico should probably be a standalone issue for discussions so see if it is of interest. On one hand this could make it possible to save all your settings in the PicoDB. On the other hand there is a fair amount of work to it and not sure how many people would actually benefit from it. If it were to be implemented I would definetly want to reuse the Firefly registered users and avoid multiple accounts confussion.

**Comment by cioraneanu** [comment]
Closed for inactivity
